api/models.py

```diff
@@ -1,54 +1,190 @@
 import datetime
 from sqlalchemy import func
+from sqlalchemy import event
+from .config import app
+
 from .config import db
 
-track_singers = db.Table('track_singers', db.metadata,
-                      db.Column('Singer_id', db.Integer, db.ForeignKey('singer.id')),
-                      db.Column('Track_id', db.Integer, db.ForeignKey('track.id')))
+
+def orm_log_after(tablename, method, owner_id, id):
+    app.logger.debug(f"user_id {owner_id} finished {method} a {tablename} {id}")
+
+
+def orm_log_before(tablename, method, owner_id):
+    app.logger.debug(f"user_id {owner_id} started {method} a {tablename}")
+
+
+track_singers = db.Table(
+    "track_singers",
+    db.metadata,
+    db.Column("Singer_id", db.Integer, db.ForeignKey("singer.id")),
+    db.Column("Track_id", db.Integer, db.ForeignKey("track.id")),
+)
 
 
 class Base(db.Model):
     __abstract__ = True
 
-    created_date = db.Column(db.DateTime, default=datetime.datetime.utcnow)
-    update_date = db.Column(db.DateTime, server_default=func.now(), onupdate=func.now())
+    created_date = db.Column(
+        db.DateTime, default=datetime.datetime.utcnow, nullable=False
+    )
+    update_date = db.Column(
+        db.DateTime, server_default=func.now(), onupdate=func.now(), nullable=False
+    )
 
 
 class Singer(Base):
-    __tablename__ = 'singer'
+    __tablename__ = "singer"
 
     id = db.Column(db.Integer, primary_key=True)
-    name = db.Column(db.String(255))
+    name = db.Column(db.String(255), nullable=False)
+    owner_id = db.Column(db.Integer, db.ForeignKey("user.id"))
 
     def __repr__(self):
-        return '<Singer %s>' % self.name
+        return "<Singer %s>" % self.name
+
+
+@event.listens_for(Singer, "after_insert")
```

Comment:
that is an interesting idea. however you'll then need to listen for rollbacks as well, otherwise you logs may be misleading